quic/s2n-quic/src/tests.rs

```diff
@@ -691,3 +704,102 @@ fn increasing_pto_count_under_loss() {
         pto_count
     );
 }
+
+// TODO: https://github.com/aws/s2n-quic/issues/1726
+//
+// The rustls tls provider is used on windows and has different
+// build options than s2n-tls. We should build the rustls provider with
+// mTLS enabled and remove the `cfg(target_os("windows"))`.
+#[cfg(not(target_os = "windows"))]
+#[test]
+fn mtls() {
+    use crate::provider::tls::default as s2n_tls;
+
+    // Ensure connection is successful under different network conditions
+    for delay_time in (500..=10_000).step_by(1000) {
+        for drop_percent in (0..=90).step_by(10) {
+            let delay_time = Duration::from_micros(delay_time);
+
+            let model = Model::default();
+            model.set_delay(delay_time);
+            // Only the Server sends a HandshakeDone frame
+            let server_subscriber = FrameSentRecorder::new();
+            let frame_sent = server_subscriber.events();
+            // Only the Client receives a HandshakeDone frame
+            let client_subscriber = FrameReceivedRecorder::new();
+            let frame_received = client_subscriber.events();
+
+            test(model.clone(), |handle| {
+                spawn(async move {
+                    // allow for 1 RTT worth of data before impairing the connection.
+                    // Both the Client and Server have received initial packets at
+                    // this point.
+                    model.set_delay(delay_time * 2);
+
+                    // simulate network impairment
+                    let drop_rate = (drop_percent / 100).into();
+                    model.set_drop_rate(drop_rate);
+                    model.set_corrupt_rate(drop_rate);
```

Comment:
Why are we setting both of these?

Reply:
I wanted to additionally impair the network, although I am not sure how much it adds to this since drop % should override corruption. I can remove.

Reply:
They're different things on the network but for QUIC, they're basically accomplishing the same thing since corruption will ultimately result in dropping the packet at the endpoint. So in this case you're actually doubling the `drop_rate` since you're applying it twice in two different manners.

Reply:
Yep that makes sense and I am seeing failures at high drop % after fixing the math.